app/Services/Questions/QuestionService.php

```diff
@@ -33,23 +33,26 @@ public function createQuestion($data)
         try {
             $folder = $this->folderRepository->firstOrCreate(['title' => $data['folder']]);
             $data['folder_id'] = $folder->id;
-            $this->tagRepository->pushCriteria(new InCriteria('title', $data['tag']));
-            $tags = $this->tagRepository->all();
-            $tmp = [];
-            foreach ($tags as $tag) {
-                $tmp[$tag->title] = $tag;
-            }
-            $tags = [];
-            foreach ($data['tag'] as $title) {
-                if (empty($tmp[$title])) {
-                    $tag = $this->tagRepository->create(['title' => $title]);
-                } else {
-                    $tag = $tmp[$title];
+            $question = $this->questionRepository->create($data);
+
+            if (!empty($data['tag'])) {
+                $this->tagRepository->pushCriteria(new InCriteria('title', $data['tag']));
+                $tags = $this->tagRepository->all();
+                $tmp = [];
+                foreach ($tags as $tag) {
+                    $tmp[$tag->title] = $tag;
+                }
+                $tags = [];
+                foreach ($data['tag'] as $title) {
+                    if (empty($tmp[$title])) {
+                        $tag = $this->tagRepository->create(['title' => $title]);
```

Comment:
Are we able to create tags in batch mode(mass insert)? Cause it's to heavy to do it one-by-one in foreach loop